verifyFiles.py

```diff
@@ -320,23 +331,26 @@ def main():
         conditions_list=read_conditions_file(conditions_file_name)
         root_dir=os.path.dirname(os.path.abspath(conditions_file_name))
         log_info("Walking through files...")
-	global checksums_flag
-        global exclude_folders
 	exclude_folders=None
 	checksums_flag = False
-	if args.checksumFile is not None:
-            checksums_flag=True
+	checksums_from_file_dict={}
 	if args.excludeFolders is not None:
 	    exclude_folders=args.excludeFolders
-        conditions_dict=get_conditions_dict(conditions_list,root_dir)
+        conditions_dict=get_conditions_dict(conditions_list,root_dir,exclude_folders)
         log_info("Checking if subject folders are missing in any condition...")
 	check_subjects(conditions_dict)
 	log_info("Checking if files are missing in any subject of any condition...")
         check_files(conditions_dict)
         log_info("Reading the metrics file...")
         metrics = read_metrics_file(args.metricsFile)
         log_info("Computing differences across subjects...")
-        diff,metric_values=n_differences_across_subjects(conditions_dict,root_dir,metrics)
+	if args.checksumFile is not None:
+            checksums_flag=True
```

Comment:
You don't need a flag *and* a file name: just read the file when the file name is available and pass the content to the methods. Methods should test whether checksums_from_file_dict is None.